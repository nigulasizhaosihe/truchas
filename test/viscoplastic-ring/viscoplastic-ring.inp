################################################################################
#
# Ring-Plug Assembly Viscoplastic Residual Stress Calculation, MTS version
# 
# This problem uses a coarse quarter symmetry mesh. The problem starts with both
# materials below their stress reference temperature. The initial stress is
# calculated and a small amount of plastic relaxation occurs. If the total time
# is increased, the assembly will continue to cool and the elastic stress and
# plastic strain will continue to evolve. This version for the test suite runs
# for a short time, exercising the initial elastic stress calculation,
# viscoplastic flow and heat conduction. This version uses the MTS model for the
# viscoplastic model
#
# - Mesh file coordinates are in meters
# - This mesh contains two materials:
#     1. Graphite plug with elastic properties (no temperature dependence)
#     2. 5754 Aluminum ring with elastic-viscoplastic properties
# - All units are SI, with temperatures in degrees K
#
# David Korzekwa, MST6
#
################################################################################


&MESH
  mesh_file = 'hytec_quarter_coarse.gen'
  interface_side_sets = 1
/

&OUTPUTS
  output_t = 0.0, 0.5 !5.0
  output_dt = 1.0
/

&NUMERICS
  dt_constant = 1e-1
  ! dt_max = 1.0
  ! dt_grow = 5.0
  ! dt_init = 1e-1
  ! dt_min = 1e-8
/

&PHYSICS
  materials = 'graphite', '5754 aluminum'
  heat_transport = .true.
  solid_mechanics = .true.
/

&DIFFUSION_SOLVER
  abs_temp_tol = 0
  rel_temp_tol = 1e-4
  abs_enthalpy_tol = 0
  rel_enthalpy_tol = 1e-4
  nlk_tol = 0.05
  verbose_stepping = .true.
/

&SOLID_MECHANICS
  maximum_iterations = 500
  abs_stress_tol = 1e-13
  abs_displ_tol = 1e100
  rel_displ_tol = 1e100
  nlk_max_vectors = 60

  ! Make the contact parameters large to model a sliding interface (no
  ! penetration or separation).
  contact_distance = 1e100
  contact_norm_trac = 1e100
/


### BODIES #####################################################################

# Plug geometry and initial conditions
&BODY
  material_name = 'graphite'
  mesh_material_number = 1
  surface_name = 'from mesh file'
  temperature = 5.50e+02
/

# Ring geometry and initial conditions
&BODY
  material_name = '5754 aluminum'
  mesh_material_number = 2
  surface_name = 'from mesh file'
  temperature = 5.50e+02
/

### BOUNDARY CONDITIONS #######################################################

# x=0, y=0, and z=0 symmetry planes
# --> insulated with zero normal displacement
&THERMAL_BC
  name = 'symmetry'
  type = 'flux'
  face_set_ids = 2, 3, 6
  flux = 0.0
/

&SOLID_MECHANICS_BC
  name = 'symmetry'
  face_set_ids = 2, 3, 6
  type = 'displacement-n'
  displacement = 0.0
/

# z = 0.0127 heat transfer and all traction components zero (default)
&THERMAL_BC
  name = 'top surface'
  type = 'htc'
  face_set_ids  = 5
  htc = 1.0e2
  ambient_temp = 2.98e+02
/

# Outer radius = 0.0508 heat transfer and all tractions zero
&THERMAL_BC
  name = 'outer radius'
  type = 'htc'
  face_set_ids = 4
  htc = 1.0e2
  ambient_temp = 2.98e+02
/

# Interface between plug and ring
&THERMAL_BC
  name = 'plug/ring interface'
  type = 'interface-htc'
  htc = 1.0e3
  face_set_ids = 1
/

&SOLID_MECHANICS_BC
  name = 'plug/ring interface'
  face_set_ids = 1
  type = 'gap-contact'
/

x&BC
  surface_name = 'from mesh file'
  mesh_surface = 1
  bc_variable = 'displacement'
  bc_type = 'normal-constraint'
  bc_value = 0.0
/

### MATERIALS ##################################################################

### GRAPHITE ###

&MATERIAL
  name = 'graphite'
  density = 1.7e3
  specific_heat = 1.925e3
  conductivity = 1.95e2
  tm_ref_density = 1.7e3
  tm_ref_temp = 800.0
  tm_linear_cte = 7.0e-6
  tm_lame1 = 3.4e+9
  tm_lame2 = 2.76e+9
/

### 5754 ALUMINUM ###

&MATERIAL
  name = '5754 aluminum'
  density = 2.7e3
  specific_heat = 8.5e2
  conductivity = 2.4e2
  tm_ref_density = 1.7e3
  tm_ref_temp = 800.0
  tm_linear_cte = 2.2e-5
  tm_lame1 = 5.20e+10
  tm_lame2 = 2.60e+10
/

&VISCOPLASTIC_MODEL
  phase = '5754 aluminum'
  model = 'MTS'
  MTS_k = 1.38e-23
  MTS_mu_0 = 28.815e9
  MTS_sig_a = 10.0e6
  MTS_d = 3.440e9
  MTS_temp_0 = 215.0
  MTS_b = 2.86e-10
  MTS_edot_0i = 1.0e7
  MTS_g_0i = 3.6
  MTS_q_i = 1.5
  MTS_p_i = 0.5
  MTS_sig_i = 107.2e6
/
