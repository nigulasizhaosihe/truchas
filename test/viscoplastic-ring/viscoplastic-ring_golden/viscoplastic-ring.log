
================================================================================

PROGRAM INFORMATION

   code:                Truchas 24.06-19-gb55f9ab8-dirty
   build architecture:  arm64
   build date/time:     2024-06-18 09:55:20
   build flags:          -g -O2 -DNDEBUG
   build host:          pn2405035.lanl.gov
   run architecture:    Darwin pn2405035.lanl.gov 23.5.0 Darwin Kernel
   run host:            pn2405035.lanl.gov
   run date/time:       27 Jun 24 18:03:00
   processors:          1

================================================================================

INPUT

Opening input file "viscoplastic-ring.inp"
Reading FUNCTION namelists ...
  none found
Reading TOOLPATH namelists ...
  none found
Reading TOOLHEAD namelists ...
  none found
Reading VFUNCTION namelists ...
  none found
Reading PHYSICS namelist ...
Reading MATERIAL namelists ...
  read namelist "graphite"
  read namelist "5754 aluminum"
Reading PHASE namelists ...
  none found
Reading PHASE_CHANGE namelists ...
  none found
Reading OUTPUTS namelist ...
Reading MESH namelist ...
Reading BODY namelists (first pass) ...
  read 2 BODY namelists
Reading BODY namelists (second pass) ...
  read 2 BODY namelists
Reading NUMERICS namelist ...
Reading SOLID_MECHANICS namelist ...
Reading VISCOPLASTIC_SOLVER namelist ...
VISCOPLASTIC_SOLVER namelist not found; using defaults.
Reading SOLID_MECHANICS_BC namelists ...
Reading VISCOPLASTIC_MODEL namelists ...
Reading DIFFUSION_SOLVER namelist ...
  using default STEPPING_METHOD value: "Adaptive BDF2"
  using default MAX_NLK_ITR value: 5
  using default MAX_STEP_TRIES value: 10
  using default MAX_NLK_VEC value: MAX_NLK_ITR - 1
  using default NLK_VEC_TOL value: 1.00E-03
  WARNING: using a pure relative error norm; temp must be bounded away from 0
  WARNING: using a pure relative error norm; enthalpy must be bounded away from 0
  using default PC_AMG_CYCLES value: 2
Reading THERMAL_BC namelists ...
  read namelist "symmetry"
  read namelist "top surface"
  read namelist "outer radius"
  read namelist "plug/ring interface"
Reading THERMAL_SOURCE namelists ...
  none found
Reading SPECIES_BC namelists ...
  none found
Reading SPECIES_SOURCE namelists ...
  none found
Reading ENCLOSURE_RADIATION namelists ...
  none found
Reading MICROSTRUCTURE namelists ...
Reading PROBE namelists ...
  none found
Input file "viscoplastic-ring.inp" closed

================================================================================

INITIALIZATION

Initializing mesh "MAIN" ...
  reading ExodusII mesh file "./hytec_quarter_coarse.gen"

Computing initial volume fractions ...
  using SM_BC[symmetry]
  using SM_BC[plug/ring interface]
SM BC faces:    227
SM BC nodes/links:    259    44
SM-C0D1 nodes:    145
SM-C0D2 nodes:     25
SM-C0D3 nodes:      1
SM-C1D0 nodes:     54
SM-C1D1 nodes:     30
SM-C1D2 nodes:      4
Nodes with requested BCs:    259    Nodes with applied BCs:    259
SM: Computing initial state...
  NLK BCE solve succeeded:    282 iterations, error,lnorm_inf =     2.655-116    9.323E-14


Initializing diffusion solver ...
  generating "scalar_cell_func1" thermal source
    none specified
  generating "scalar_cell_func2" thermal source
    none specified
  generating "interface-htc" thermal interface condition
    using THERMAL_BC[plug/ring interface]
  generating "gap-radiation" thermal interface condition
    none specified
  generating "flux" thermal boundary condition
    using THERMAL_BC[symmetry]
  generating "oriented-flux" thermal boundary condition
    none specified
  generating "htc" thermal boundary condition
    using THERMAL_BC[top surface]
    using THERMAL_BC[outer radius]
  generating "radiation" thermal boundary condition
    none specified
  generating "temperature" thermal boundary condition
    none specified
  diffusion solver initialized

Computing consistent initial state for HT/SD solver ...
  computing consistent face temperatures and radiosities ...
  ||Rface(4)||= 4.60E-06, ||Rface(0)||= 7.18E+00, ||ΔTface||_max= 1.82E-05

Computing consistent initial state derivative for HT/SD solver ...
  computing consistent face temperatures and radiosities ...
  ||Rface(4)||= 3.47E-05, ||Rface(0)||= 1.05E+01, ||ΔTface||_max= 1.61E-04

Initializing solution probes ...

================================================================================

EXECUTION

          1: t =   0.00000E+00, dt(constant) =   1.00000E-01

DS: dt=1.000E-01, NFUN:NPC=0000002:00001, NNR:NNF:NSR=0000:0000:0000
  NLK BCE solve succeeded:    300 iterations, error,lnorm_inf =     4.020-116    8.444E-14

          2: t =   1.00000E-01, dt(constant) =   1.00000E-01

DS: dt=1.000E-01, NFUN:NPC=0000005:00001, NNR:NNF:NSR=0000:0000:0000
  NLK BCE solve succeeded:    224 iterations, error,lnorm_inf =     1.793-116    7.132E-14

          3: t =   2.00000E-01, dt(constant) =   1.00000E-01

DS: dt=1.000E-01, NFUN:NPC=0000008:00001, NNR:NNF:NSR=0000:0000:0000
  NLK BCE solve succeeded:    280 iterations, error,lnorm_inf =     1.744-115    8.460E-14

          4: t =   3.00000E-01, dt(constant) =   1.00000E-01

DS: dt=1.000E-01, NFUN:NPC=0000011:00001, NNR:NNF:NSR=0000:0000:0000
  NLK BCE solve succeeded:    268 iterations, error,lnorm_inf =     1.493-115    8.672E-14

          5: t =   4.00000E-01, dt(constant) =   1.00000E-01

DS: dt=1.000E-01, NFUN:NPC=0000014:00001, NNR:NNF:NSR=0000:0000:0000
  NLK BCE solve succeeded:    252 iterations, error,lnorm_inf =     2.328-115    8.209E-14

================================================================================

TERMINATION
                 Final Time:  5.0000E-01 after     5 steps


TIMING SUMMARY
--------------
  4.861E+00 --- Total
    5.480E-03 --- Input
    1.878E-01 --- Initialization
      1.434E-03 --- mesh-MAIN
        4.600E-05 --- ghost-cells
          4.400E-05 --- cell-ghosts
            2.600E-05 --- node-neighbors
            4.000E-06 --- ghost-cells
          1.000E-06 --- link-ghosts
          1.000E-06 --- assemble-ghosts
      1.010E-04 --- VOF Initialize
      1.816E-01 --- solid mechanics
        6.000E-05 --- properties
        4.400E-04 --- compute_precon
          4.380E-04 --- precon-compute
            4.290E-04 --- residual
              3.570E-04 --- strain
              7.200E-05 --- stress
              0.000E+00 --- BCs
        1.108E-01 --- compute_f
          1.107E-01 --- residual
            8.709E-02 --- strain
            2.216E-02 --- stress
            1.270E-03 --- BCs
        1.197E-03 --- lnorm
        9.570E-04 --- apply_precon
          4.060E-04 --- precon-apply
        6.341E-02 --- nka
        2.220E-04 --- du_norm
      9.100E-04 --- viscoplasticity
      2.530E-04 --- HTSD function
        2.430E-04 --- HT function
      7.000E-06 --- ssor-setup
      1.150E-04 --- ssor-solve
    4.667E+00 --- Main Cycle
      1.100E-05 --- Time Step
      3.631E-03 --- Diffusion Solver
        1.070E-03 --- HTSD precon compute
          4.800E-05 --- hypre-matrix-copy
          8.800E-04 --- boomer-setup
        4.340E-04 --- HTSD function
          3.680E-04 --- HT function
        1.778E-03 --- HTSD precon apply
          1.775E-03 --- HT precon apply
            1.708E-03 --- boomer-solve
      4.657E+00 --- solid mechanics
        3.130E-04 --- properties
        1.607E-02 --- compute_precon
          1.607E-02 --- precon-compute
            1.602E-02 --- residual
              1.688E-03 --- strain
              1.387E-02 --- viscoplasticity
                9.097E-03 --- integrate
              4.580E-04 --- stress
              3.000E-06 --- BCs
        4.334E+00 --- compute_f
          4.334E+00 --- residual
            4.350E-01 --- strain
            3.757E+00 --- viscoplasticity
              2.526E+00 --- integrate
            1.355E-01 --- stress
            5.822E-03 --- BCs
        5.496E-03 --- lnorm
        4.532E-03 --- apply_precon
          1.851E-03 --- precon-apply
        2.908E-01 --- nka
        1.077E-03 --- du_norm


truchas terminated normally on 27 Jun 24 at 18:03:05
