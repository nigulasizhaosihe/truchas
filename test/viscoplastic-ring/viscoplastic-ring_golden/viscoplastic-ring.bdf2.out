
TRAP step      1: T= 0.000000000000000E+00, H= 1.000000000000000E-01
  Preconditioner updated at T= 1.000000000000000E-01
    1: error= 1.055243896304403E+00
    2: error= 3.280154471665362E-02
  NLK BCE solve succeeded:   2 iterations, error= 3.280154471665362E-02
  Step accepted: no local error control

BDF2 step      2: T= 1.000000000000000E-01, H= 1.000000000000000E-01
    1: error= 4.832743526356986E-01
    2: error= 1.256512175729720E-01
    3: error= 2.427478087009423E-03
  NLK BCE solve succeeded:   3 iterations, error= 2.427478087009423E-03
  Step accepted: no local error control

BDF2 step      3: T= 2.000000000000000E-01, H= 1.000000000000000E-01
    1: error= 4.434716546611286E-01
    2: error= 1.170006387107462E-01
    3: error= 2.051689692810854E-03
  NLK BCE solve succeeded:   3 iterations, error= 2.051689692810854E-03
  Step accepted: no local error control

BDF2 step      4: T= 3.000000000000000E-01, H= 9.999999999999998E-02
    1: error= 3.003417478485521E-01
    2: error= 8.111518239097137E-02
    3: error= 1.364900706909945E-03
  NLK BCE solve succeeded:   3 iterations, error= 1.364900706909945E-03
  Step accepted: perr= 3.769577397251471E-01

BDF2 step      5: T= 4.000000000000000E-01, H= 9.999999999999998E-02
    1: error= 1.817957892045701E-01
    2: error= 5.043350928078025E-02
    3: error= 8.379024518664317E-04
  NLK BCE solve succeeded:   3 iterations, error= 8.379024518664317E-04
  Step accepted: perr= 2.293131021009367E-01
